From 2613e4dd6fe7b351581799e11980104b25e443f1 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Thu, 12 Sep 2019 20:34:28 +0200
Subject: [PATCH 5/8] Most horrific: Remove ramdisk's zygote init scripts

This is needed because:
- only secilc is run soon enough in /system to
remove it
- placing an init.zygote in system won't have init replace it, it's the
first that appears that wins

Change-Id: I8be31ceb9ef2124d04994d9fb08fc8012a2f819e
---
 secilc/secilc.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/secilc/secilc.c b/secilc/secilc.c
index 631b4b0a..3dd6e7ce 100644
--- a/secilc/secilc.c
+++ b/secilc/secilc.c
@@ -34,6 +34,8 @@
 #include <getopt.h>
 #include <sys/stat.h>
 
+#include <unistd.h>
+
 #ifdef ANDROID
 #include <cil/cil.h>
 #else
@@ -121,6 +123,11 @@ int main(int argc, char *argv[])
 	};
 	int i;
 
+	unlink("/init.zygote32.rc");
+	unlink("/init.zygote64_32.rc");
+	unlink("/init.zygote64.rc");
+	unlink("/init.zygote32_64.rc");
+
 	while (1) {
 		opt_char = getopt_long(argc, argv, "o:f:U:hvt:M:PDmNc:GX:", long_opts, &opt_index);
 		if (opt_char == -1) {
-- 
2.25.1

